explorer/src/context/main.tsx

```diff
@@ -1,17 +1,33 @@
+import { PaletteMode } from '@mui/material';
 import * as React from 'react';
+import { GatewayResponse, MixNodeResponse, ValidatorsResponse } from 'src/typeDefs/node-status-api-client';
+// import { MixNodeResponse, GatewayResponse } from 'src/typeDefs/node-status-api-client';
 import { Api } from '../api';
 
-export const MainContext = React.createContext({});
+type NodeApiResponse = {
+  data: MixNodeResponse | GatewayResponse | ValidatorsResponse | number | null
+  error: string | null
+}
+interface State {
+  mode: PaletteMode
+  toggleMode: () => void
+  mixnodes: NodeApiResponse | null
+  gateways: NodeApiResponse | null
+  validators: NodeApiResponse | null
+  block: NodeApiResponse | null
+};
+
+export const MainContext = React.createContext({} as State);
```

Comment:
Create context has a generic parameter that defines the context's type.

```suggestion
export const MainContext = React.createContext<State>({});
```

Reply:
This yields a type error - `Argument of type '{}' is not assignable to parameter of type 'State'.
  Type '{}' is missing the following properties from type 'State': mode, toggleMode`
  
  So i've given it what i'm assuming is a default state of mode 'dark' and changed toggleMode to optional.